x-pack/plugins/maps/server/mvt/get_tile.ts

```diff
@@ -238,6 +330,59 @@ export async function getTile({
           props[FEATURE_ID_PROPERTY_NAME] = features[i].id;
         }
       }
+
+      const counts = countVectorShapeTypes(features);
+
+      const fieldNames = new Set<string>();
+      features.forEach((feature) => {
+        for (const key in feature.properties) {
+          if (feature.properties.hasOwnProperty(key) && key !== 'key' && key !== 'gridCentroid') {
+            fieldNames.add(key);
+          }
+        }
+      });
+
+      const fieldMeta: FieldMeta = {};
+      fieldNames.forEach((fieldName: string) => {
+        if (
```

Comment:
How about moving these checks up to the if statement on line 339 so all key validation occurs in the same place